### DISCUSSION TITLE
Graph: Improve when to render time regions for time ranges that do not fully surround the region

### DISCUSSION THREAD

**Discussion by fty4** [grafana/grafana#37022] [Dashboards, Panels & Visualizations]
**What happened**:
The time region will not be displayed on a small time range e.g. `now-6h -> now` (or smaller):
![image](https://user-images.githubusercontent.com/16336640/77511180-cd6bad80-6e70-11ea-972d-e1f45088208a.png)

If I select a larger time range is selected (for me it is greater than 6h) the time region will be displayed:
![image](https://user-images.githubusercontent.com/16336640/77511233-f2f8b700-6e70-11ea-9a15-aae4da607278.png)

_Note: I have a timezone offset of +1_

**What you expected to happen**:
The time regions will be always displayed - doesn't matter which time range is selected.

**How to reproduce it (as minimally and precisely as possible)**:

1. create a new panel
1. add any query - e.g. time()
1. add a time region - e.g. from 6 pm to 6 am
1. select different time ranges

**Anything else we need to know?**:
no - thats all (:

**Environment**:
- Grafana version: Grafana v6.7.1 (ca6d08d5cb) via docker image, also appeared on v6.6.1
- Data source type & version: prometheus v2.14.0
- OS Grafana is installed on: SUSE Linux Enterprise Server 12 SP5  (x86_64), Docker version 19.03.5, build 633a0ea838f1
- User OS & Browser: Windows 10, Firefox 74.0 x64 // Windows 10, Chrome Version 80.0.3987.149 (Offizieller Build) (64-Bit) // Ubuntu 18.0.4.4 LTS, Firefox 74.0 x64
- Grafana plugins: -
- Others: -

**Comment by martinberg** [comment]
I'm having this issue with 6.7.4 aswell.
I was able to narrow it down to time region will not be displayed if the starting point of the region is 2 hours or more out of bounds for the graph.
The same behavior doesn't happen if I only show the start of the time region and +2 hours.

Missing time region
![Without_Time_Region](https://user-images.githubusercontent.com/9891659/84121158-0f030200-aa37-11ea-8df5-3872ca3470f1.png)

Containing time region
![With_Time_Region](https://user-images.githubusercontent.com/9891659/84121161-10342f00-aa37-11ea-87f6-7cbe840f2a80.png)

My timezone is +2 hours therefore the graph ends at 00:00 and not 22:00 as entered in the settings for the time region.

**Comment by torkelo** [comment]
Changed title and added help wanted label, if anyone wants to improve this feature

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
